src/api/v1/users/volunteer_logs.ts

```diff
@@ -0,0 +1,298 @@
+  };
+}
+
+
+const routes: Hapi.ServerRoute[] = [
+
+  {
+    method: 'GET',
+    path: '/users/me/volunteer-logs',
+    options: {
+      description: 'Read own volunteer logs',
+      auth: {
+        strategy: 'standard',
+        access: {
+          scope: ['volunteer_logs-parent:read'],
+        },
+      },
+      validate: {
+        query: { since, until, ...query },
+      },
+      response: { schema: response },
+      pre: [
+        { method: getCommunityBusiness, assign: 'communityBusiness' },
+      ],
+    },
+    handler: async (request: GetMyVolunteerLogsRequest, h) => {
+      const {
+        server: { app: { knex } },
+        auth: { credentials: { user } },
+        pre: { communityBusiness },
+        query,
+      } = request;
+
+      const since = new Date(query.since);
+      const until = new Date(query.until);
+
+      const logs = await VolunteerLogs.fromUserAtCommunityBusiness(
+        knex,
+        user,
+        communityBusiness,
+        { since, until, limit: query.limit, offset: query.offset }
+      );
+
+      return Promise.all(logs.map(VolunteerLogs.serialise));
+    },
+  },
+
+  {
+    method: 'POST',
+    path: '/users/me/volunteer-logs',
+    options: {
+      description: 'Create volunteer log for own user',
+      auth: {
+        strategy: 'standard',
+        access: {
+          scope: ['volunteer_logs-parent:write'],
+        },
+      },
+      validate: {
+        payload: {
+          organisationId: id,
+          activity: Joi.string().required(),
+          duration: Joi.object({
+            hours: Joi.number().integer().min(0),
+            minutes: Joi.number().integer().min(0),
+            seconds: Joi.number().integer().min(0),
+          }).required(),
+          startedAt: Joi.date().iso().max('now').default(() => new Date().toISOString(), 'now'),
+        },
+      },
+      response: { schema: response },
+      pre: [
+        { method: getCommunityBusiness, assign: 'communityBusiness' },
+      ],
+    },
+    handler: async (request: PostMyVolunteerLogsRequest, h) => {
+      const {
+        server: { app: { knex } },
+        auth: { credentials: { user } },
+        pre: { communityBusiness },
+        payload,
+      } = request;
+
+      if (payload.organisationId && communityBusiness.id !== payload.organisationId) {
+        return Boom.badRequest('Cannot create log for different organisation');
+      }
+
+      try {
+        const log = await VolunteerLogs.add(knex, {
+          ...payload,
+          organisationId: communityBusiness.id,
+          userId: user.id,
+        });
+
+        return VolunteerLogs.serialise(log);
+
+      } catch (error) {
+        if (error.code === '23502') { // Violation of null constraint implies invalid activity
+          return Boom.badRequest('Invalid activity');
+        }
+        throw error;
+      }
+    },
+  },
+
+  {
+    method: 'GET',
+    path: '/users/me/volunteer-logs/{logId}',
```

Comment:
🚚 `/users/me/volunteer-logs/:id` -> `/community-businesses/me/volunteer-logs/:id`

Reply:
I don't think this one should move. If we can access the entire collection through `GET /users/volunteers/me/volunteer-logs` it follows that we can also access individual resources through `GET /users/volunteers/me/volunteer-logs/:id`

Reply:
IMO I don't think that any logs should be accessible under the `/users` endpoint. Logs are owned by cbs and should be accessed as such.

Reply:
On second look, I don't think that any logs should be accessible under the /users endpoint. Logs are owned by cbs and should be accessed as such.